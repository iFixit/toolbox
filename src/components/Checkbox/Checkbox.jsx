import React from 'react';
import PropTypes from 'prop-types';

import CheckboxRadio from '../../internal/CheckboxRadio';

const Checkbox = props => <CheckboxRadio type="checkbox" {...props} />;

Checkbox.propTypes = {
   /** Checkbox is selected. */
   checked: PropTypes.bool,
   /** Set class name of containing element. */
   className: PropTypes.string,
   /** Indicates that the form control is not available for interaction. */
   disabled: PropTypes.bool,
   /** Represents a caption for the form control. */
   label: PropTypes.string,
   /** The value of the control. */
   value: PropTypes.string,
   /** Callback when the control is toggled. */
   onChange: PropTypes.func,
   /** Callback when mouse enters component. */
   onMouseEnter: PropTypes.func,
   /** Callback when mouse leaves component. */
   onMouseLeave: PropTypes.func,
};

Checkbox.defaultProps = {
   checked: false,
   className: '',
   disabled: false,
   label: '',
   value: 'on',
   onChange: () => {},
   onMouseEnter: () => {},
   onMouseLeave: () => {},
};

export default Checkbox;
