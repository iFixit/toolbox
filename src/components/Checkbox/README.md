```
<Checkbox label="Cole" checked />
```

```
<Checkbox label="Cole" />
```


```
<Checkbox label="Cole" checked disabled />
```


```
<Checkbox label="Cole" disabled />
```