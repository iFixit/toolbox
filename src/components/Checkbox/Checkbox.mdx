---
name: Checkbox
route: /components/checkbox
menu: Components
---

import Checkbox from './Checkbox'
import { Playground, PropsTable } from 'docz'

# Checkbox

## Examples

### Checked

<Playground>
  <Checkbox label="Checkbox label" checked />
</Playground>

### Unchecked

<Playground>
  <Checkbox label="Checkbox label" />
</Playground>

### Disabled checked

<Playground>
  <Checkbox label="Checkbox disabled" checked disabled />
</Playground>

### Disabled unchecked

<Playground>
  <Checkbox label="Checkbox disabled" disabled />
</Playground>

### Invalid unchecked

<Playground>
  <Checkbox required label="Checkbox invalid" showInvalid={true} />
</Playground>

### Event callbacks

<Playground>
  <Checkbox
    label="Checkbox label"
    onChange={({ checked, value }) => console.log(checked, value)}
    onMouseEnter={() => console.log('mouse entered')}
    onMouseLeave={() => console.log('mouse left')}
  />
</Playground>

## Props

<PropsTable of={Checkbox} />
