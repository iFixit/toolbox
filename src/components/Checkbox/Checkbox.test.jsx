import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

test('renders without crashing', () => {
   const tree = renderer.create(<Checkbox />).toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with label', () => {
   const tree = renderer.create(<Checkbox label="Test Label" />).toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with children', () => {
   const tree = renderer
      .create(
         <Checkbox>
            Normal Text
            <strong>Strong</strong>
         </Checkbox>,
      )
      .toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with custom margin', () => {
   const tree = renderer
      .create(<Checkbox label="Test Label" labelMargin="100px" />)
      .toJSON();
   expect(tree).toMatchSnapshot();
});
