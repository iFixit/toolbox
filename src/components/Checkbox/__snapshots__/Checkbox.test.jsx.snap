// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with children 1`] = `
.c3 {
  display: inline-block;
  width: 24px;
  height: 24px;
  color: currentColor;
  stroke-width: 2;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  font-size: 16px;
  color: rgba(0,3,6,0.87);
}

.c1 {
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  width: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
}

.c4 {
  margin-left: 16px;
  line-height: 1.25;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  box-sizing: content-box;
  width: 16px;
  height: 16px;
  line-height: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  color: transparent;
  background-color: #FFFFFF;
  border: 1px solid rgba(0,3,6,0.26);
  border-radius: 4px;
}

<label
  className="c0"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInvalid={[Function]}
    type="checkbox"
    value="on"
  />
  <span
    className="c2 c3"
    color="currentColor"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"inherit\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-check\\"><polyline points=\\"20 6 9 17 4 12\\"></polyline></svg>",
      }
    }
    size={24}
    strokeWidth={2}
  />
  <span
    className="c4"
  >
    Normal Text
    <strong>
      Strong
    </strong>
  </span>
</label>
`;

exports[`renders with custom margin 1`] = `
.c3 {
  display: inline-block;
  width: 24px;
  height: 24px;
  color: currentColor;
  stroke-width: 2;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  font-size: 16px;
  color: rgba(0,3,6,0.87);
}

.c1 {
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  width: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
}

.c4 {
  margin-left: 100px;
  line-height: 1.25;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  box-sizing: content-box;
  width: 16px;
  height: 16px;
  line-height: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  color: transparent;
  background-color: #FFFFFF;
  border: 1px solid rgba(0,3,6,0.26);
  border-radius: 4px;
}

<label
  className="c0"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInvalid={[Function]}
    type="checkbox"
    value="on"
  />
  <span
    className="c2 c3"
    color="currentColor"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"inherit\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-check\\"><polyline points=\\"20 6 9 17 4 12\\"></polyline></svg>",
      }
    }
    size={24}
    strokeWidth={2}
  />
  <span
    className="c4"
  >
    Test Label
  </span>
</label>
`;

exports[`renders with label 1`] = `
.c3 {
  display: inline-block;
  width: 24px;
  height: 24px;
  color: currentColor;
  stroke-width: 2;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  font-size: 16px;
  color: rgba(0,3,6,0.87);
}

.c1 {
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  width: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
}

.c4 {
  margin-left: 16px;
  line-height: 1.25;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  box-sizing: content-box;
  width: 16px;
  height: 16px;
  line-height: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  color: transparent;
  background-color: #FFFFFF;
  border: 1px solid rgba(0,3,6,0.26);
  border-radius: 4px;
}

<label
  className="c0"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInvalid={[Function]}
    type="checkbox"
    value="on"
  />
  <span
    className="c2 c3"
    color="currentColor"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"inherit\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-check\\"><polyline points=\\"20 6 9 17 4 12\\"></polyline></svg>",
      }
    }
    size={24}
    strokeWidth={2}
  />
  <span
    className="c4"
  >
    Test Label
  </span>
</label>
`;

exports[`renders without crashing 1`] = `
.c3 {
  display: inline-block;
  width: 24px;
  height: 24px;
  color: currentColor;
  stroke-width: 2;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  font-size: 16px;
  color: rgba(0,3,6,0.87);
}

.c1 {
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  width: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
}

.c4 {
  margin-left: 16px;
  line-height: 1.25;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  box-sizing: content-box;
  width: 16px;
  height: 16px;
  line-height: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: all 150ms ease-in-out;
  transition: all 150ms ease-in-out;
  color: transparent;
  background-color: #FFFFFF;
  border: 1px solid rgba(0,3,6,0.26);
  border-radius: 4px;
}

<label
  className="c0"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    checked={false}
    className="c1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onInvalid={[Function]}
    type="checkbox"
    value="on"
  />
  <span
    className="c2 c3"
    color="currentColor"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"inherit\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-check\\"><polyline points=\\"20 6 9 17 4 12\\"></polyline></svg>",
      }
    }
    size={24}
    strokeWidth={2}
  />
  <span
    className="c4"
  >
    
  </span>
</label>
`;
